pkg/minikube/cni/cni.go

```diff
@@ -200,41 +205,105 @@ func applyManifest(cc config.ClusterConfig, r Runner, f assets.CopyableFile) err
+// - failed to clean up sandbox container "..." network for pod "...": networkPlugin cni failed to teardown pod "..." network: missing network name]
+// It is caller's responsibility to restart container runtime for these changes to take effect.
+func ConfigureLoopback(r Runner) error {
+	loopback := "/etc/cni/net.d/*loopback.conf*" // usually: 200-loopback.conf
+	// turn { "cniVersion": "0.3.1", "type": "loopback" }
+	// into { "cniVersion": "0.3.1", "name": "loopback", "type": "loopback" }
+	if _, err := r.RunCmd(exec.Command("sh", "-c", fmt.Sprintf("stat %s", loopback))); err != nil {
+		klog.Warningf("%q not found, skipping patching loopback config step", loopback)
+		return nil
+	}
+	if _, err := r.RunCmd(exec.Command(
+		"sudo", "find", filepath.Dir(loopback), "-maxdepth", "1", "-type", "f", "-name", filepath.Base(loopback), "-exec", "sh", "-c",
+		`grep -q loopback {} && ( grep -q name {} || sudo sed -i '/"type": "loopback"/i \ \ \ \ "name": "loopback",' {} ) && sudo sed -i 's|"cniVersion": ".*"|"cniVersion": "1.0.0"|g' {}`, ";")); err != nil {
+		return fmt.Errorf("unable to patch loopback config %q: %v", loopback, err)
+	}
+	return nil
+}
+
+// ConfigureDefaultBridgeCNIs configures all default bridge CNIs on a node (designated by runner).
+// If network plugin is set (could be, eg "cni" or "kubenet"), it will disable all default bridges by changing extension to "mk_disabled" of *bridge* config file(s) found in /etc/cni/net.d to avoid conflicts.
+// Otherwise, it will change ip address range to match DefaultPodCIDR in all *bridge* config file(s) found in /etc/cni/net.d.
+// It's usually called before deploying new CNI and on node restarts, to avoid conflicts and flip-flopping of pods' ip addresses.
+// It is caller's responsibility to restart container runtime for these changes to take effect.
+// ref: https://github.com/containernetworking/cni/blob/main/libcni/conf.go
+// ref: https://kubernetes.io/docs/tasks/administer-cluster/migrating-from-dockershim/troubleshooting-cni-plugin-related-errors/
+func ConfigureDefaultBridgeCNIs(r Runner, networkPlugin string) error {
+	if networkPlugin != "" {
+		return disableAllBridgeCNIs(r)
+	}
+	return configureAllBridgeCNIs(r, DefaultPodCIDR)
+}
+
+func disableAllBridgeCNIs(r Runner) error {
+	path := "/etc/cni/net.d"
+
+	out, err := r.RunCmd(exec.Command(
+		"sudo", "find", path, "-maxdepth", "1", "-type", "f", "-name", "*bridge*", "-not", "-name", "*.mk_disabled", "-printf", "%p, ", "-exec", "sh", "-c",
+		`sudo mv {} {}.mk_disabled`, ";"))
+	if err != nil {
+		return fmt.Errorf("failed to disable all bridge cni configs in %q: %v", path, err)
+	}
+	configs := strings.Trim(out.Stdout.String(), ", ")
+	if len(configs) == 0 {
+		klog.Infof("no bridge cni configs found in %q - nothing to disable", path)
+		return nil
+	}
+	klog.Infof("disabled [%s] bridge cni config(s)", configs)
+
+	return nil
+}
+
+func configureAllBridgeCNIs(r Runner, cidr string) error {
```

Comment:
please add a comment for this func

Reply:
a detailed comment is with the calling func ConfigureDefaultBridgeCNIs() just above this one, but sure - i can move those comments down to be with this func

Reply:
i've moved and added some more comments around the three related functions